FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +21,48 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 	
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		
+		feedLoaderCompletion = completion
+		client.get(from: url) { [weak self] (result) in
+			
+			switch result {
+			
+			case .success((let data, let httpResponse)):
+				
+				guard httpResponse.statusCode == 200 else{
+					self?.feedLoaderCompletion?(.failure(Error.invalidData))
+					return
+				}
+				
+				guard let responseData = try? JSONDecoder().decode(FeedImageItemsStruct.self, from: data) else {
+					self?.feedLoaderCompletion?(.failure(Error.invalidData))
+					return
+				}
```

Comment:
Since both guard statements complete with `.failure(Error.invalidData)` before returning, the expression can be combined into a single guard:

```suggestion
guard httpResponse.statusCode == 200,
	let responseData = try? JSONDecoder().decode(FeedImageItemsStruct.self, from: data) else {
	completion(.failure(Error.invalidData))
	return
}
```

Reply:
Thanks, Mike, I will update likewise